packages/app/src/cli/services/dev/processes/dev-session/dev-session.ts

```diff
@@ -61,12 +62,34 @@ export class DevSession {
 
   /**
    * Handle the app event, after validating the event it might trigger a dev session update.
+   * If an update is already in progress, it will queue the event to be processed after
+   * the current update completes.
    * @param event - The app event
    */
   private async onEvent(event: AppEvent) {
     const eventIsValid = await this.validateAppEvent(event)
     if (!eventIsValid) return
-    this.currentBundleController?.abort()
+
+    this.appEventsQueue.push(event)
+
+    if (!this.updateInProgress) await this.processEvents()
```

Comment:
This is going to block the caller while we wait for all events to process -- which may be slow if more come in whilst processing is happening. I don't think the `await` is really needed.